src/shared/VacationTable/VacationTable.tsx

```diff
@@ -96,32 +102,29 @@ function TablePaginationActions(props: TablePaginationActionsProps) {
   );
 }
 
-function createData(
-  id: number,
-  name: string,
-  calories: number,
-  fat: number,
-  notes: number,
-  action: number
-) {
-  return { id, name, calories, fat, notes, action };
-}
-
-const rows: any = [
-  //   createData(1, "Frozen yoghurt", 159, 6.0, 24, 4.0),
-  //   createData(2, "Ice cream sandwich", 237, 9.0, 37, 4.3),
-  //   createData(3, "Eclair", 262, 16.0, 24, 6.0),
-  //   createData(4, "Cupcake", 305, 3.7, 67, 4.3),
-  //   createData(5, "Gingerbread", 356, 16.0, 49, 3.9),
-  //   createData(6, "Gingerbreads", 356, 16.0, 49, 3.9),
-];
-
 const VacationTable: FC = () => {
   const [page, setPage] = React.useState<number>(0);
   const [rowsPerPage, setRowsPerPage] = React.useState<number>(5);
+  const { data, setEdit, setData } = useContext(UserData);
+  const navigate = useNavigate();
+  const { user } = useAuth();
+
+  useEffect(() => {
+    const getDataFromStorage = JSON.parse(
+      localStorage.getItem("data") as string
+    );
+    const user = JSON.parse(localStorage.getItem("user") as string);
```

Comment:
I think you can create util functions for set/get data for the localStore and pass there just a key.